--- bfd/elfxx-ba.c	2014-04-04 10:22:08.000000000 +0300
+++ bfd/elfxx-ba.c	2019-03-27 12:32:36.000000000 +0200
@@ -3040,7 +3040,7 @@
   struct eh_frame_sec_info *sec_info;
   struct eh_cie_fde *fde;
   struct eh_cie_fde *fde_end;
-  Elf_Internal_Rela *irels, *irel, *irels_end;
+  Elf_Internal_Rela *irels, *irel = NULL, *irels_end;
   Elf_Internal_Rela **irel_arr, **irel_itt, **irel_last, **irel_end;
   bfd_byte *contents;
   bfd_byte *fde_contents, *prev_al_end;
@@ -3541,7 +3541,7 @@
 {
   struct elf_ba_section_data *sd = sec->used_by_bfd;
   Elf_Internal_Shdr *symtab_hdr;
-  Elf_Internal_Rela **irel_arr, **irel, **irelend;
+  Elf_Internal_Rela **irel_arr, **irel = NULL, **irelend;
   Elf_Internal_Rela **last_irel;
   bfd_byte *contents;
   Elf_Internal_Sym *isymbuf = NULL;
