--- ftdi_driver.c	2019-04-17 17:35:21.064316026 +0300
+++ ftdi_driver.c	2019-04-17 17:38:22.224515619 +0300
@@ -77,17 +77,19 @@
 
 /* vendor, product, name pairs of supported devices */
 struct ftdi_device_desc supported_devs[] = {
-    { 0x0403, 0x6010, "Beyond", "Debug Key", l_beyond_debug_key_v1_2, 0 },          // Beyond Debug Key
-    { 0x0403, 0x6010, "Beyond", "Debug Hub", l_beyond_debug_key_v1_1, 0 },          //   (eng. sample 2)
-    { 0x0403, 0x6010, "Beyond", "JTAG Adapter", l_beyond_debug_key_v1_0, 0 },       //   (eng. sample 1)
-    { 0x0403, 0xcff8, "Amontec", NULL, l_amontec_jtagkey2, 0 },                     // Amontec jtagkey/jtagkey2
-    { 0x15ba, 0x002a, NULL, NULL, l_olimex_armusb, 0 },                             // Olimex arm-usb-tiny-h
-    { 0x15ba, 0x002b, NULL, NULL, l_olimex_armusb, 0 },                             // Olimex arm-usb-ocd-h"
-    { 0x15ba, 0x0004, NULL, NULL, l_olimex_armusb, 0 },                             // Olimex arm-usb-tiny"
-    { 0x15ba, 0x0003, NULL, NULL, l_olimex_armusb, 0 },                             // Olimex arm-usb-ocd"
-    { 0x0403, 0x6010, "Digilent", "Digilent Adept USB Device", l_digilent_hs1, 0 }, // Digilent hs1 ("adept")
-    { 0x0403, 0xac09, "Atomic", NULL, l_amontec_jtagkey, 0 },                       // Atomic programming AP-114 (same layout as Amontec)
-    { 0x0403, 0x8220, "DISTORTEC", "JTAG-lock-pick Tiny 2", l_lockpick2, 1 },       // JTAG-lock-pick Tiny 2
+    { 0x0403, 0x6010, "Beyond", "Debug Key", l_beyond_debug_key_v1_2, 0 },                // Beyond Debug Key
+    { 0x0403, 0x6010, "Beyond", "Debug Hub", l_beyond_debug_key_v1_1, 0 },                //   (eng. sample 2)
+    { 0x0403, 0x6010, "Beyond", "JTAG Adapter", l_beyond_debug_key_v1_0, 0 },             //   (eng. sample 1)
+    { 0x0403, 0xcff8, "Amontec", NULL, l_amontec_jtagkey2, 0 },                           // Amontec jtagkey/jtagkey2
+    { 0x15ba, 0x002a, NULL, NULL, l_olimex_armusb, 0 },                                   // Olimex arm-usb-tiny-h
+    { 0x15ba, 0x002b, NULL, NULL, l_olimex_armusb, 0 },                                   // Olimex arm-usb-ocd-h"
+    { 0x15ba, 0x0004, NULL, NULL, l_olimex_armusb, 0 },                                   // Olimex arm-usb-tiny"
+    { 0x15ba, 0x0003, NULL, NULL, l_olimex_armusb, 0 },                                   // Olimex arm-usb-ocd"
+    { 0x0403, 0x6010, "Digilent", "Digilent Adept USB Device", l_digilent_hs1, 0 },       // Digilent hs1 ("adept")
+    { 0x0403, 0xac09, "Atomic", NULL, l_amontec_jtagkey, 0 },                             // Atomic programming AP-114 (same layout as Amontec)
+    { 0x0403, 0x8220, "DISTORTEC", "JTAG-lock-pick Tiny 2", l_lockpick2, 1 },             // JTAG-lock-pick Tiny 2
+    { 0x0403, 0x8a98, "TIAO", "TIAO USB Multi-Protocol Adapter", l_amontec_jtagkey2, 0 }, // TIAO Tumpa
+    { 0x0403, 0x6014, NULL, NULL, l_amontec_jtagkey2, 0 },             			  // generic FT232H board
     { 0, 0 }
 };

